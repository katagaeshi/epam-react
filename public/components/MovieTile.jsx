import React from 'react';
import PropTypes from 'prop-types';

const MovieTile = props => (
  <div>
    <img src={props.src} alt="movie" />
    <span>{props.title}</span>
    <span>{props.year}</span>
    <span>{props.genre}</span>
  </div>
);

MovieTile.propTypes = {
  src: PropTypes.string,
  title: PropTypes.string.isRequired,
  year: PropTypes.number,
  genre: PropTypes.string,
};

MovieTile.defaultProps = {
  src: null,
  year: null,
  genre: 'unknown',
};

export default MovieTile;
