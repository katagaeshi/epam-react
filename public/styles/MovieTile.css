.MovieTile {
  display: grid;
}

.MovieTile > img {
  object-fit: contain;
  max-width: 100%;
  grid-row: 1 / 1;
  grid-column: 1 / 3;
}

.ReleaseDate {
  grid-row: 2 / 2;
  grid-column: 2 / 2;
  text-align: right;
}

.ReleaseDate > span {
  border-style: solid;
  border-width: 1px;
  display: inline-block;
  white-space: nowrap;
}

.Genre {
  grid-row: 3 / 3;
  grid-column: 1 / 1;
}

.Title {
  grid-row: 2 / 2;
  grid-column: 1 / 1;
  font-weight: bold;
}
