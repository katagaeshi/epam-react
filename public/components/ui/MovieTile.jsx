import React from 'react';
import PropTypes from 'prop-types';
import '../../styles/MovieTile.css';

const MovieTile = props => (
  <div className="MovieTile">
    <img data-id={props.id} src={props.poster_path} alt="movie" />
    <span className="Title">{props.title}</span>
    <div className="ReleaseDate">
      <span>{props.release_date}</span>
    </div>
    <span className="Genre">{props.genres.toString()}</span>
  </div>
);

MovieTile.propTypes = {
  id: PropTypes.string.isRequired,
  poster_path: PropTypes.string,
  title: PropTypes.string.isRequired,
  release_date: PropTypes.string,
  genres: PropTypes.arrayOf(PropTypes.string),
};

MovieTile.defaultProps = {
  poster_path: null,
  release_date: null,
  genres: 'unknown',
};

export default MovieTile;
