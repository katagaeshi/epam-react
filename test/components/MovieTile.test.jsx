import React from 'react';
import renderer from 'react-test-renderer';
import MovieTile from './../../public/components/MovieTile';

describe('MovieTile', () => {
  it('Snapshot', () => {
    const tree = renderer
      .create(<MovieTile
        src="www.example.com"
        title="Pokemon"
        year={2001}
        genre="Anime"
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
