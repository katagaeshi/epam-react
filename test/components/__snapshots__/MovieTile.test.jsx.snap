// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTile Snapshot 1`] = `
<div>
  <img
    alt="movie"
    src="www.example.com"
  />
  <span>
    Pokemon
  </span>
  <span>
    2001
  </span>
  <span>
    Anime
  </span>
</div>
`;
